google_photos_takeout_helper/__main__.py

```diff
@@ -246,15 +248,95 @@ def set_file_exif_date(file, creation_date):
         try:
             _piexif.insert(_piexif.dump(exif_dict), file)
         except Exception as e:
-            print('Couldnt insert exif!')
+            print("Couldn't insert exif!")
             print(e)
 
-
     def get_date_str_from_json(json):
         return _datetime.fromtimestamp(
             int(json['photoTakenTime']['timestamp'])
         ).strftime('%Y:%m:%d %H:%M:%S')
 
+    def change_to_rational(number):
+        """convert a number to rantional
+        Keyword arguments: number
+        return: tuple like (1, 2), (numerator, denominator)
+        """
+        f = Fraction(str(number))
+        return f.numerator, f.denominator
+
+    # got this here https://github.com/hMatoba/piexifjs/issues/1#issuecomment-260176317
+    def degToDmsRational(degFloat):
+        min_float = degFloat % 1 * 60
+        sec_float = min_float % 1 * 60
+        deg = math.floor(degFloat)
+        deg_min = math.floor(min_float)
+        sec = round(sec_float * 100)
+
+        return [(deg, 1), (deg_min, 1), (sec, 100)]
+
+    def set_file_geo_data(file, json):
+        """
+        Reads the geoData from google and saves it to the EXIF. This works assuming that the geodata looks like -100.12093, 50.213143. Something like that.
+
+        Written by DalenW.
+        :param file:
+        :param json:
+        :return:
+        """
+
+        exif_dict = _piexif.load(file)
+
+        longitude = float(json['geoDataExif']['longitude'])
+        latitude = float(json['geoDataExif']['latitude'])
+        altitude = json['geoDataExif']['altitude']
```

Comment:
Can we prioritise geoData set from Photos editor? I personally have many photos that I received from Messenger from my friends, and have manually set their location through GPhotos, but it didn't edit their exif's - because they didn't have it at all

```suggestion
        longitude = float(json['geoData']['longitude'])
        latitude = float(json['geoData']['latitude'])
        altitude = json['geoData']['altitude']
        # Prioritise geoData set from GPhotos editor
        if longitude == 0 and latitude == 0:
            longitude = float(json['geoDataExif']['longitude'])
            latitude = float(json['geoDataExif']['latitude'])
            altitude = json['geoDataExif']['altitude']
```

Reply:
Didn't even know that was a thing.